sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala

```diff
@@ -415,7 +505,12 @@ object TypeSig {
   val DATE: TypeSig = new TypeSig(TypeEnum.ValueSet(TypeEnum.DATE))
   val TIMESTAMP: TypeSig = new TypeSig(TypeEnum.ValueSet(TypeEnum.TIMESTAMP))
   val STRING: TypeSig = new TypeSig(TypeEnum.ValueSet(TypeEnum.STRING))
-  val DECIMAL: TypeSig = new TypeSig(TypeEnum.ValueSet(TypeEnum.DECIMAL))
+  val DECIMAL_64: TypeSig =
+    new TypeSig(TypeEnum.ValueSet(TypeEnum.DECIMAL), DType.DECIMAL64_MAX_PRECISION)
+  val DECIMAL_128_NO_OVERFLOW: TypeSig =
+    new TypeSig(TypeEnum.ValueSet(TypeEnum.DECIMAL), DecimalType.MAX_PRECISION - 1)
+  val DECIMAL_128_FULL: TypeSig =
```

Comment:
I'm a bit confused on the semantics for these, and I think a comment could help.  For example, is DECIMAL_128_NO_OVERFLOW covering cases where we need to check and prevent silent overflow or where we don't need to check?

Reply:
I thought about removing it entirely until we are ready to start using it. But then `DECIMAL_128_FULL` felt a little odd.  I will clean it up.